File: Objective-C/AdvancedExample/AdvancedExample/Video.m
#import "Video.h"

@implementation Video

- (instancetype)initWithTitle:(NSString *)title
                    thumbnail:(UIImage *)thumbnail
                        video:(NSString *)video
                          tag:(NSString *)tag {
  self = [super init];
  if (self) {
    _title = [title copy];
    _thumbnail = [thumbnail copy];
    _video = [video copy];
    _tag = [tag copy];
  }
  return self;
}

@end

File: Swift/AdvancedExample/AdvancedExample/Video.swift
import Foundation
import UIKit

class Video: NSObject {

  // The title of the video.
  var title: String

  // The thumbnail for the video list.
  var thumbnail: UIImage

  // The URL for the video media file.
  var video: String

  //The URL for the VAST response.
  var tag: String

  // Returns an initialized video.
  init(title: String, thumbnail: UIImage!, video: String, tag: String) {
    self.title = title
    self.thumbnail = thumbnail
    self.video = video
    self.tag = tag
  }

}
